components/lsp-form/LongShort.tsx

```diff
@@ -35,6 +36,56 @@ const LongShort: FC<Props> = ({
   shortTokenBalance,
   shortTokenDecimals,
 }) => {
+  const setCollateralInput = useCallback(
+    (
+      tokenAmount: string,
+      valueCallback: React.Dispatch<React.SetStateAction<string>>
+    ) => {
+      const normalizedCPP = ethers.utils.formatEther(collateralPerPair);
+
+      const newAmount = Number(tokenAmount) * Number(normalizedCPP);
+
+      setAmount(newAmount.toString());
+      valueCallback(tokenAmount);
+    },
+    [collateralPerPair]
+  );
+
+  const maxTokensRedeemable = useCallback(() => {
+    const normalizedCPP = ethers.utils.formatEther(collateralPerPair);
+
+    let ltb = "0",
+      stb = "0",
+      am = "0";
+
+    if (longTokenBalance.gte(shortTokenBalance)) {
+      ltb = ethers.utils.formatUnits(shortTokenBalance, shortTokenDecimals);
+      stb = ethers.utils.formatUnits(shortTokenBalance, shortTokenDecimals);
+      am = (
+        Number(
+          ethers.utils.formatUnits(shortTokenBalance, shortTokenDecimals)
+        ) * Number(normalizedCPP)
+      ).toString();
+    }
+
+    if (longTokenBalance.lt(shortTokenBalance)) {
```

Comment:
fortunately the decimals in these cases will always be the saame, otherwise u would have to normalizze the decimals before this comparison

Reply:
It looks like they are always the same as the collateral too, so I think I can actually simplify some of the props in the code.